
License identified for Makefile:
# SPDX-License-Identifier: GPL-2.0

License identified for arch/x86/Kconfig:
# SPDX-License-Identifier: GPL-2.0

License identified for kernel/Makefile:
# SPDX-License-Identifier: GPL-2.0

--- a/Makefile.orig	2023-09-09 00:03:56.050164268 -0700
+++ b/Makefile	2023-09-09 00:04:55.810166125 -0700
@@ -685,6 +685,9 @@ endif # KBUILD_EXTMOD
 # Defaults to vmlinux, but the arch makefile usually adds further targets
 all: vmlinux
 
+CFLAGS_PGO_CLANG := -fprofile-generate
+export CFLAGS_PGO_CLANG
+
 CFLAGS_GCOV	:= -fprofile-arcs -ftest-coverage
 ifdef CONFIG_CC_IS_GCC
 CFLAGS_GCOV	+= -fno-tree-loop-im
--- a/arch/x86/Kconfig.orig	2023-09-09 00:03:45.058163927 -0700
+++ b/arch/x86/Kconfig	2023-09-09 00:05:36.450167387 -0700
@@ -108,6 +108,7 @@ config X86
 	select ARCH_SUPPORTS_KMAP_LOCAL_FORCE_MAP	if NR_CPUS <= 4096
 	select ARCH_SUPPORTS_LTO_CLANG
 	select ARCH_SUPPORTS_LTO_CLANG_THIN
+	select ARCH_SUPPORTS_PGO_CLANG		if X86_64
 	select ARCH_USE_BUILTIN_BSWAP
 	select ARCH_USE_MEMTEST
 	select ARCH_USE_QUEUED_RWLOCKS
--- a/kernel/Makefile.orig	2023-09-09 00:03:55.250164243 -0700
+++ b/kernel/Makefile	2023-09-09 00:06:21.918168800 -0700
@@ -117,6 +117,7 @@ obj-$(CONFIG_SHADOW_CALL_STACK) += scs.o
 obj-$(CONFIG_HAVE_STATIC_CALL) += static_call.o
 obj-$(CONFIG_HAVE_STATIC_CALL_INLINE) += static_call_inline.o
 obj-$(CONFIG_CFI_CLANG) += cfi.o
+obj-$(CONFIG_PGO_CLANG) += pgo/
 
 obj-$(CONFIG_PERF_EVENTS) += events/
 
